### ISSUE TITLE
WASM Migration Phase 8: Implement safeZoneAdjustment() in Rust/WASM for all activation functions

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-TS#1140]
## Parent Issue
Part of #1136 - Replace all JS squash functions with Rust/WASM

## Background
Following Phase 7 (unSquash), this phase implements `safeZoneAdjustment()` in WASM. This function determines how suitable it is to backpropagate through a neuron based on saturation levels. It's critical for training stability.

## Scope
Implement `safeZoneAdjustment()` in Rust/WASM for all activation functions.

### Understanding safeZoneAdjustment
The function returns a float from 0 (not safe) to 1 (fully safe) indicating how useful it is to backpropagate through a neuron:
- **1.0**: Fully in safe zone, gradient flows freely
- **0.0**: Completely saturated, no gradient should flow
- **0.0-1.0**: Partial safety, used for gradual fade-out

### Functions to implement
Each activation has different safe zones based on their saturation behaviour:

| Function | Safe Zone (Full) | Fade Zone | Notes |
|----------|------------------|-----------|-------|
| Identity | (-∞, ∞) = 1.0 | None | Never saturates |
| ReLU | [0, ∞) = 1.0 | x < 0: recovery | Dead ReLU recovery |
| ReLU6 | (0, 6) = 1.0 | Edges | Both ends saturate |
| LeakyReLU | (-∞, ∞) = 1.0 | None | Never fully saturates |
| SELU/ELU | [-2, ∞) = 1.0 | x < -2: fade | Negative saturation |
| Logistic | [-6, 6] = full | ±6 to ±10 fade | Classic sigmoid |
| Tanh | [-2, 2] = full | ±2 to ±6 fade | Similar to logistic |
| HardTanh | (-1, 1) = 1.0 | At ±1 = 0 | Hard boundaries |
| Softsign | [-5, 5] = full | Fade to ±10 | Slow saturation |
| Softplus | [-10, ∞) = full | x < -10 fade | One-sided |
| Swish/Mish | Similar to tanh | | |
| GELU | Similar to ReLU | | Smoother transition |
| Sine/Cosine | Always varying | 1.0 | Periodic |
| Tan | Avoid ±π/2 | | Asymptotes |
| ArcTan | (-∞, ∞) | Fade at extremes | |
| Gaussian | [-3, 3] = full | Fade beyond | Bell curve |
| BentIdentity | (-∞, ∞) = 1.0 | None | Never saturates |
| BipolarSigmoid | [-6, 6] = full | Fade | |
| Bipolar/Step | 0.0 always | None | Discontinuous |
| Others | Per function | | |

### Implementation approach
1. Add `safe_zone_adjustment(squash_type: SquashType, raw_input: f32, error: f32) -> f32` in lib.rs
2. Implement recovery logic for each function (when error direction helps recovery)
3. Export via wasm_bindgen
4. Add TypeScript wrapper `wasmSafeZoneAdjustment(squashType: SquashType, rawInput: number, error: number): number`
5. Add tests comparing WASM and JS implementations

### Key patterns from JS implementations
```rust
// Pattern for bounded functions (like Tanh):
if raw_input >= safe_low && raw_input <= safe_high {
    return 1.0;  // Fully safe
}
// Recovery direction
if raw_input < safe_low && error > 0.0 {
    return 0.2;  // Allow small recovery gradient
}
// Gradual fade
if raw_input > safe_high && raw_input <= max {
    return 1.0 - (raw_input - safe_high) / (max - safe_high);
}
return 0.0;  // Fully saturated
```

### Acceptance Criteria
- [ ] All activation functions have safeZoneAdjustment implemented in Rust
- [ ] Recovery logic implemented for each function
- [ ] Exported via wasm_bindgen
- [ ] TypeScript wrapper function added
- [ ] Tests verify WASM and JS results match
- [ ] All existing tests pass
- [ ] `./quality.sh` passes

## Notes
- The recovery logic (allowing small gradients when error direction would help unsaturate) is critical for training
- Non-finite inputs should return 0
- Discontinuous functions (Bipolar, Step) should always return 0
